File: internal/routex/entity/pricing.go
package entity

import "time"

type PricingSetting struct {
	ID               int    `gorm:"primaryKey;autoIncrement:false"`
	ETag             string `gorm:"size:64;not null"`
	PlatformCurrency string `gorm:"size:3;not null"`
}
type PricingExchangeRate struct {
	Currency string `gorm:"primaryKey;size:3"`
	Rate     string `gorm:"size:40;not null"`
}
type ModelPrice struct {
	ID               string `gorm:"primaryKey;size:30"`
	ProviderModelID  string `gorm:"size:30;not null;uniqueIndex"`
	ContextThreshold int64  `gorm:"not null"`
	UpdateSource     string `gorm:"size:20;not null"`
	FollowRepository bool   `gorm:"not null"`
	CreatedAt        time.Time
	UpdatedAt        time.Time
}
type PriceRate struct {
	ID           string `gorm:"primaryKey;size:30"`
	ModelPriceID string `gorm:"size:30;not null"`
	Metric       string `gorm:"size:30;not null"`
	Tier         string `gorm:"size:20;not null"`
	Unit         string `gorm:"size:20;not null"`
	Currency     string `gorm:"size:3;not null"`
	Amount       string `gorm:"size:40;not null"`
	Enabled      bool   `gorm:"not null"`
}

File: website/src/types/pricing.ts
export const metrics = [
  'INPUT_TOKEN',
  'OUTPUT_TOKEN',
  'CACHE_READ_TOKEN',
  'CACHE_WRITE_TOKEN',
] as const
export const currencies = ['USD', 'CNY', 'EUR', 'GBP', 'JPY', 'HKD', 'SGD'] as const
export type PriceMetric = (typeof metrics)[number]
export type PriceCurrency = (typeof currencies)[number]
export interface PriceRate {
  id?: string
  metric: PriceMetric
  tier: 'base' | 'long_context'
  unit: '1M_TOKEN'
  currency: PriceCurrency
  amount: string
  enabled: boolean
}
export interface ModelPrice {
  id: string
  provider_model_id: string
  provider_id: string
  upstream_name: string
  protocol: string
  context_threshold: 0 | 128000 | 200000
  update_source: string
  follow_repository: boolean
  rates: PriceRate[]
}
export interface PricePage {
  etag: string
  currency: { platform_currency: PriceCurrency; rates: Record<string, string> }
  items: ModelPrice[]
  next_cursor?: string
}
export interface PriceWrite {
  etag: string
  items: {
    provider_model_id: string
    context_threshold: ModelPrice['context_threshold']
    rates: Omit<PriceRate, 'id'>[]
  }[]
}
